aws/data_source_aws_ecr_lifecycle_policy_document.go

```diff
@@ -0,0 +1,201 @@
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+)
+
+type EcrLifecyclePolicyDoc struct {
+	Rules []*EcrLifecyclePolicyStatement `json:"rules"`
+}
+
+type EcrLifecyclePolicyStatement struct {
+	RulePriority int                                     `json:"rulePriority,omitempty"`
+	Description  string                                  `json:"description,omitempty"`
+	Selection    EcrLifecyclePolicyStatementSelectionSet `json:"selection,omitempty"`
+	Action       EcrLifecyclePolicyAction                `json:"action"`
+}
+
+type EcrLifecyclePolicySelection struct {
+	TagStatus     string        `json:"tagStatus,omitempty"`
+	TagPrefixList []interface{} `json:"tagPrefixList,omitempty"`
+	CountType     string        `json:"countType,omitempty"`
+	CountUnit     string        `json:"countUnit,omitempty"`
+	CountNumber   int           `json:"countNumber,omitempty"`
+}
+
+type EcrLifecyclePolicyAction struct {
+	Type string `json:"type"`
+}
+
+type EcrLifecyclePolicyStatementSelectionSet EcrLifecyclePolicySelection
+
+func dataSourceAwsEcrLifecyclePolicyDocument() *schema.Resource {
+
+	return &schema.Resource{
+		Read: dataSourceAwsEcrLifecyclePolicyDocumentRead,
+
+		Schema: map[string]*schema.Schema{
+			"rule": {
+				Type:     schema.TypeList,
+				Required: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"priority": {
+							Type:     schema.TypeInt,
+							Required: true,
+						},
+						"description": {
+							Type:     schema.TypeString,
+							Required: false,
+							Optional: true,
+						},
+						"selection": {
+							Type:     schema.TypeSet,
+							Required: false,
+							Optional: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"tag_status": {
+										Type:         schema.TypeString,
+										Required:     false,
+										Optional:     true,
+										Default:      "any",
+										ValidateFunc: validation.StringInSlice([]string{"tagged", "untagged", "any"}, false),
+									},
+									"tag_prefix_list": {
+										Type:     schema.TypeList,
+										Required: false,
+										Optional: true,
+										Elem:     &schema.Schema{Type: schema.TypeString},
+									},
+									"count_type": {
+										Type:         schema.TypeString,
+										Required:     false,
+										Optional:     true,
+										ValidateFunc: validation.StringInSlice([]string{"imageCountMoreThan", "sinceImagePushed", "any"}, false),
+									},
+									"count_unit": {
+										Type:         schema.TypeString,
+										Required:     false,
+										Optional:     true,
+										ValidateFunc: validation.StringInSlice([]string{"days"}, false),
+									},
+									"count_number": {
+										Type:     schema.TypeInt,
+										Required: false,
+										Optional: true,
+									},
+								},
+							},
+						},
+						"action": {
+							Type:     schema.TypeSet,
```

Comment:
This attribute should be `Type: schema.TypeList` and `MaxItems: 1` 👍

Reply:
Fixed